homeassistant/components/xiaomi_miio/select.py

```diff
@@ -0,0 +1,253 @@
+        self._device_features = 0
+        self._state_attrs = {}
+        self._controller = selector
+        self._current_option = None
+        self._enum_class = None
+        self._attributes = None
+        self._available_attributes = []
+
+    @property
+    def options(self):
+        """Return available options."""
+        return self._controller.options
+
+    @property
+    def available(self):
+        """Return true when state is known."""
+        return super().available and self._available
+
+    @property
+    def current_option(self):
+        """Return the current option."""
+        if not self.available and self.coordinator.data:
+            self._available = True
+            self._current_option = self._extract_value_from_attribute(
+                self.coordinator.data, self._controller.short_name
+            )
+        return (
+            getattr(self, SERVICE_TO_METHOD[self._controller.service]["property"])
+            if self.available
+            else None
+        )
+
+    @staticmethod
+    def _extract_value_from_attribute(state, attribute):
+        value = getattr(state, attribute)
+        if isinstance(value, Enum):
+            return value.value
+
+        return value
+
+    async def async_select_option(self, option: str) -> None:
+        """Set an option of the miio device."""
+        if option not in self.options:
+            _LOGGER.warning(
+                "Selection '%s' is not a valid '%s'", option, self._controller.name
+            )
+            return
+        method = getattr(self, SERVICE_TO_METHOD[self._controller.service]["method"])
+        await method(option)
+
+
+class XiaomiAirHumidifierSelector(XiaomiSelector):
+    """Representation of a Xiaomi Air Humidifier selector."""
+
+    def __init__(self, name, device, entry, unique_id, controller, coordinator):
+        """Initialize the plug switch."""
+        super().__init__(name, device, entry, unique_id, controller, coordinator)
+        if self._model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
+            self._device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA_AND_CB
+        elif self._model in [MODEL_AIRHUMIDIFIER_CA4]:
+            self._device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA4
+        else:
+            self._device_features = FEATURE_FLAGS_AIRHUMIDIFIER
+
+        self._state_attrs.update(
+            {attribute: None for attribute in self._available_attributes}
+        )
+
+    @callback
+    def _handle_coordinator_update(self):
+        """Fetch state from the device."""
+        # On state change the device doesn't provide the new state immediately.
+        state = self.coordinator.data
+        if not state:
+            return
+        _LOGGER.debug("Got new state: %s", state)
+        self._available = True
+        self._current_option = self._extract_value_from_attribute(
+            state, self._controller.short_name
+        )
+        self.async_write_ha_state()
```

Comment:
I also suggest `self._available = True` in `XiaomiCoordinatedMiioEntity.__init__()`.
```suggestion
        self._available = True
        self._current_option = self._extract_value_from_attribute(
            self.coordinator.data, self._controller.short_name
        )
        self.async_write_ha_state()
```